server/src/main/kotlin/io/spine/examples/pingh/server/LocalServer.kt

```diff
@@ -0,0 +1,132 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.server
+
+import io.ktor.client.engine.cio.CIO
+import io.spine.environment.DefaultMode
+import io.spine.examples.pingh.clock.IntervalClock
+import io.spine.examples.pingh.github.ClientId
+import io.spine.examples.pingh.github.ClientSecret
+import io.spine.examples.pingh.github.GitHubApp
+import io.spine.examples.pingh.github.of
+import io.spine.examples.pingh.mentions.RemoteGitHubSearch
+import io.spine.examples.pingh.mentions.newMentionsContext
+import io.spine.examples.pingh.sessions.RemoteGitHubAuthentication
+import io.spine.examples.pingh.sessions.RemoteGitHubUsers
+import io.spine.examples.pingh.sessions.newSessionsContext
+import io.spine.server.Server
+import io.spine.server.ServerEnvironment
+import io.spine.server.delivery.Delivery
+import io.spine.server.storage.memory.InMemoryStorageFactory
+import io.spine.server.transport.memory.InMemoryTransportFactory
+import java.util.Properties
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * The port on which the Pingh server runs.
+ */
+private const val pinghPort = 50051
+
+/**
+ * The entry point for the local server application.
+ *
+ * Server setup and startup process:
+ *
+ * 1. Configures the environment for local use, which includes loading GitHub App secrets
+ * from the configuration file.
+ *
+ * 2. Initializes an [IntervalClock] to emit an event with the current time to the server
+ * every second.
+ *
+ * 3. Builds and launches the Pingh server.
+ */
+public fun main() {
+    configureEnvironment()
+    val gitHubApp = loadGitHubAppSecrets()
+    val server = createServer(gitHubApp)
+    IntervalClock(1.seconds).start()
+    server.start()
+    server.awaitTermination()
+}
+
+/**
+ * Configures the server environment.
+ *
+ * Server side of this application is currently running in in-memory storage mode.
+ * Therefore, any changes made by users of this application will not be persisted
+ * in-between the application launches.
+ */
+private fun configureEnvironment() {
+    ServerEnvironment
+        .`when`(DefaultMode::class.java)
+        .use(InMemoryStorageFactory.newInstance())
+        .use(Delivery.localAsync())
+        .use(InMemoryTransportFactory.newInstance())
+}
+
+/**
+ * Loads GitHub application secrets from resource folder.
+ */
+private fun loadGitHubAppSecrets(): GitHubApp {
+    val properties = Properties()
+    val path = "/local/config/server.properties"
+    Application::class.java.getResourceAsStream(path).use {
+        properties.load(it)
+    }
+    check(properties.containsKey("github-app.client.id")) {
```

Comment:
Let's generify the implementation to get rid of the repeating pieces of code and string constants.